Readability.js

```diff
@@ -1266,23 +1270,25 @@ Readability.prototype = {
     e = e || this._doc;
     if (!e)
       return;
-    var cur = e.firstChild;
 
-    // Remove any root styles, if we're able.
-    if (typeof e.removeAttribute === 'function' && e.className !== 'readability-styled')
-      e.removeAttribute('style');
-
-    // Go until there are no more child nodes
-    while (cur !== null) {
-      if (cur.nodeType === cur.ELEMENT_NODE) {
-        // Remove style attribute(s) :
-        if (cur.className !== "readability-styled")
-          cur.removeAttribute("style");
+    if (e.nodeType === e.ELEMENT_NODE &&
+        e.namespaceURI === 'http://www.w3.org/1999/xhtml' &&
```

Comment:
Hm, there's a fallback to `this._doc`, but we pass in `articleContent`, which is never the document, and there's only 1 callsite, so perhaps we should just remove the fallback?

For SVG, I mean, the `width` and `height` would already not be removed because we're only doing that for the elements in a list that you already created. 
The `style` point is well-made, but is an existing bug, right? We could try to refine that later. I don't really have good ideas about that anyway, short of either creating more lists of elements or trying to teach JSDOMParser about namespaces (but really, don't try to do that... namespaces are seriously messed up). The simplest thing might just be to prevent the recursion entirely for `<svg>` elements, and leave it at that. That would miss out HTML-nested-in-SVG, but I think we can live with that... if people put table markup in their SVG, I don't really want to know!